fs2-google-pubsub-http/src/main/scala/com/permutive/pubsub/producer/http/PubsubHttpProducerConfig.scala

```diff
@@ -2,11 +2,27 @@ package com.permutive.pubsub.producer.http
 
 import scala.concurrent.duration._
 
+/**
+  * Configuration for the PubSub HTTP producer.
+  *
+  * @param host                              host of PubSub
+  * @param port                              port of PubSub
+  * @param isEmulator                        whether the target PubSub is an emulator or not
+  * @param oauthTokenRefreshInterval         how often to refresh the Google OAuth token
+  * @param onTokenRefreshError               what to do if the token refresh fails
+  * @param oauthTokenFailureRetryDelay       initial delay for retrying OAuth token retrieval
+  * @param oauthTokenFailureRetryNextDelay   next delay for retrying OAuth token retrieval
+  * @param oauthTokenFailureRetryMaxAttempts how many times to attempt
```

Comment:
Maybe worth clarifying that it will throw the last error once this is reached?

Reply:
`.lastOrError` actually just throws an `NoSuchElementException`, rather than the "last error", here's the implementation of `lastOfError`:
```scala
    def lastOrError(implicit G: MonadError[G, Throwable]): G[O] =
      last.flatMap(_.fold(G.raiseError(new NoSuchElementException): G[O])(G.pure))
```

Reply:
But FS2 will have raised raised an error already (which is the last encountered) having reached the limit of retries.